src/Common/MoistThermodynamics/states.jl

```diff
@@ -176,28 +194,70 @@ end
 
 Constructs a [`PhaseNonEquil`](@ref) thermodynamic state from:
 
- - `θ_liq_ice` - liquid-ice potential temperature
- - `ρ` - density
- - `q_pt` - phase partition
+ - `θ_liq_ice` liquid-ice potential temperature
+ - `ρ` (moist-)air density
+ - `q_pt` phase partition
+and, optionally
+ - `tol` tolerance for non-linear equation solve
+ - `maxiter` maximum iterations for non-linear equation solve
 """
-function LiquidIcePotTempSHumNonEquil(θ_liq_ice::FT, ρ::FT, q_pt::PhasePartition{FT}) where {FT<:Real}
-    T = air_temperature_from_liquid_ice_pottemp_non_linear(θ_liq_ice, ρ, q_pt)
+function LiquidIcePotTempSHumNonEquil(θ_liq_ice::FT,
+                                      ρ::FT,
+                                      q_pt::PhasePartition{FT},
+                                      tol::FT=FT(1e-3),
+                                      maxiter::Int=10) where {FT<:Real}
+    T = air_temperature_from_liquid_ice_pottemp_non_linear(θ_liq_ice, ρ, tol, maxiter, q_pt)
     e_int = internal_energy(T, q_pt)
-    return PhaseNonEquil(e_int, ρ, q_pt)
+    return PhaseNonEquil{FT}(e_int, ρ, q_pt)
 end
 
 """
     LiquidIcePotTempSHumNonEquil_given_pressure(θ_liq_ice, p, q_pt)
 
 Constructs a [`PhaseNonEquil`](@ref) thermodynamic state from:
 
- - `θ_liq_ice` - liquid-ice potential temperature
- - `p` - pressure
- - `q_pt` - phase partition
+ - `θ_liq_ice` liquid-ice potential temperature
+ - `p` pressure
+ - `q_pt` phase partition
 """
 function LiquidIcePotTempSHumNonEquil_given_pressure(θ_liq_ice::FT, p::FT, q_pt::PhasePartition{FT}) where {FT<:Real}
     T = air_temperature_from_liquid_ice_pottemp_given_pressure(θ_liq_ice, p, q_pt)
     ρ = air_density(T, p, q_pt)
     e_int = internal_energy(T, q_pt)
-    return PhaseNonEquil(e_int, ρ, q_pt)
+    return PhaseNonEquil{FT}(e_int, ρ, q_pt)
+end
+
+"""
+    tested_convergence_range(FT)
+
+A range of input arguments to thermodynamic state constructors
+ - `e_int` internal energy
+ - `ρ` (moist-)air density
+ - `q_tot` total specific humidity
+ - `q_pt` phase partition
+ - `T` air temperature
+ - `θ_liq_ice` liquid-ice potential temperature
+that are tested for convergence in saturation adjustment.
+
+Note that the output vectors are of size ``n^3``, and they
+should span the input arguments to all of the constructors.
+"""
+function tested_convergence_range(FT, n)
+  ρ_range  = range(FT(1e-3),  stop=FT(1.6), length=n)
+  RH_range = range(FT(0),     stop=FT(1.02), length=n)
```

Comment:
I am concerned that the maximum of `RH_range`is so small that we are not effectively testing saturation adjustment. For `n<50`, we are never testing the nonlinear saturation adjustment here. (Note that saturation adjustment is only tested for `RH > 1`.) I'd prefer a larger maximum, and maybe widening test tolerances so that tests do not fail.

Reply:
Fixed!